app/src/main/java/ru/practicum/android/diploma/ui/root/RootActivity.kt

```diff
@@ -142,5 +146,23 @@ class RootActivity : AppCompatActivity() {
 //            }
         }
     }
-
+// функция проверяет, находится ли точка касания внутри области видимости текстового поля.
+    // Если точка касания находится вне области видимости текстового поля,
+    // то функция снимает фокус с текстового поля и скрывает клавиатуру.
+
+    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
+        if (event.action == MotionEvent.ACTION_DOWN) {
+            val v = currentFocus
+            if (v is TextInputEditText) {
+                val outRect = Rect()
+                v.getGlobalVisibleRect(outRect)
+                if (!outRect.contains(event.rawX.toInt(), event.rawY.toInt())) {
+                    v.clearFocus()
+                    val imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
+                    imm.hideSoftInputFromWindow(v.getWindowToken(), 0)
+                }
+            }
+        }
+        return super.dispatchTouchEvent(event)
+    }
```

Comment:
```suggestion
            val inputMethodManager =
                requireActivity().getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
            inputMethodManager?.hideSoftInputFromWindow(binding.searchEditText.windowToken, 0)
            binding.searchEditText.clearFocus()
```
В плеере делали так, почему этот способ не подходит?